index.js

```diff
@@ -82,18 +94,30 @@ class Pipe2Jpeg extends Transform {
           this._size += sliced.length;
           this._jpeg = Buffer.concat(this._chunks, this._size);
           this._chunks = [];
+          this._lastSize = this._size;
           this._size = 0;
           this._sendJpeg();
           if (pos === chunkLength) {
             break;
           }
         }
       } else {
+        if (this._lastByte === _SOI[0] && chunk[0] === _SOI[1]) {
+          // SOI was split across chunks
+          const startByte = Buffer.from([_SOI[0]]);
+          chunk = Buffer.concat([startByte, chunk]);
+          chunkLength += 1;
+          pos = Math.max(pos - 1, 0);
+        }
         const soi = chunk.indexOf(_SOI, pos);
         if (soi === -1) {
+          // save the last byte in case the soi is broken across chunks
+          this._lastByte = chunk[chunkLength - 1];
           break;
         } else {
-          pos = soi + this._increment;
+          // as an optimization, jump forward half of the previous jpeg size
+          const stepForward = this._lastSize / 2;
```

Comment:
yes, good catch! I'm pretty sure lastSize will be divisible by two because usually the smallest block size will be an even number of bytes, but it doesn't hurt to add this check